### PR TITLE
make register.GroupVersion use GroupVersion

### PR DESCRIPTION
**Pull request by deads2k**
Switches the register package to use GroupVersions.

@liggitt @caesarxuchao

### CODE REVIEW

File: pkg/api/registered/registered.go [hunk 1]
```diff
@@ -22,46 +22,60 @@ import (
...
-			glog.Fatalf("invalid api version: %s in KUBE_API_VERSIONS: %s. List of valid API versions: %v",
-				version, os.Getenv("KUBE_API_VERSIONS"), validGroupVersions)
+	userRequestedVersions := os.Getenv("KUBE_API_VERSIONS")
+	if len(userRequestedVersions) != 0 {
+		// reset the supported versions
+		supportedVersions = []unversioned.GroupVersion{}
+		for _, version := range strings.Split(userRequestedVersions, ",") {
+			gv, err := unversioned.ParseGroupVersion(version)
+			if err != nil {
+				panic(err)
```
Comment (liggitt) [inline pkg/api/registered/registered.go]: glog.Fatalf, like below?

**Comment by liggitt** [comment]
one nit on panic vs glog.Fatalf, LGTM otherwise

**Comment by liggitt** [comment]
travis failure looks related:
```
-      --storage-versions="componentconfig/v1alpha1,extensions/v1beta1,metrics/v1alpha1,v1": The versions to store resources with. Different groups may be stored in different versions. Specified in the format "group1/version1,group2/version2...". This flag expects a complete list of storage versions of ALL groups registered in the server. It defaults to a list of preferred versions of all registered groups, which is derived from the KUBE_API_VERSIONS environment variable.

+      --storage-versions="componentconfig/v1alpha1,extensions/v1beta1,v1": The versions to store resources with. Different groups may be stored in different versions. Specified in the format "group1/version1,group2/version2...". This flag expects a complete list of storage versions of ALL groups registered in the server. It defaults to a list of preferred versions of all registered groups, which is derived from the KUBE_API_VERSIONS environment variable.
```

File: pkg/api/registered/registered.go [hunk 2]
```diff
@@ -22,46 +22,60 @@ import (
...
-		"metrics/v1alpha1":         true,
+	validGroupVersions := map[unversioned.GroupVersion]bool{
+		unversioned.GroupVersion{Group: "", Version: "v1"}:                      true,
+		unversioned.GroupVersion{Group: "extensions", Version: "v1beta1"}:       true,
+		unversioned.GroupVersion{Group: "componentconfig", Version: "v1alpha1"}: true,
+		unversioned.GroupVersion{Group: "metrics", Version: "v1alpha1"}:         true,
 	}
 
 	// The default list of supported api versions, in order of most preferred to the least.
-	defaultSupportedVersions := "v1,extensions/v1beta1,componentconfig/v1alpha1"
```
Comment (liggitt) [inline pkg/api/registered/registered.go]: old default didn't include metrics, new one does... not sure if that was a bug before

File: pkg/api/install/install.go [hunk 3]
```diff
@@ -52,16 +51,16 @@ func init() {
 	registeredGroupVersions := registered.GroupVersionsForGroup("")
 	groupVersion := registeredGroupVersions[0]
 	*groupMeta = latest.GroupMeta{
-		GroupVersion: groupVersion,
-		Group:        apiutil.GetGroup(groupVersion),
-		Version:      apiutil.GetVersion(groupVersion),
-		Codec:        runtime.CodecFor(api.Scheme, groupVersion),
+		GroupVersion: groupVersion.String(),
```
Comment (caesarxuchao) [inline pkg/api/install/install.go]: I'll plumb the GroupVersion to the `latest` package so the conversion here can be spared.

**Comment by caesarxuchao** [comment]
ref #17216

**Comment by liggitt** [comment]
LGTM. @k8s-bot test this

**Comment by deads2k** [comment]
@liggitt I'm not seeing a build log from the k8s-bot.  Personal problem on my part?

**Comment by liggitt** [comment]
> @liggitt I'm not seeing a build log from the k8s-bot. Personal problem on my part?

no, the bot has been weird... not sure who owns artifact collector

**Comment by deads2k** [comment]
clean rebase, no changes.  re-adding lgtm

**Comment by caesarxuchao** [comment]
@k8s-oncall this PR is not picked up by the submit queue.

**Comment by wojtek-t** [comment]
@caesarxuchao - we had huge problems with merging over last few days; Jenkins was mostly red over last 3 days

**Comment by caesarxuchao** [comment]
@wojtek-t Thanks. Will the submit-queue pick up this one automatically or do I need to rerun the tests?

**Comment by wojtek-t** [comment]
@caesarxuchao - it should be picked automatically; although (since merge-bot is starting with the oldest PRs), it won't be on the top of the queue...

**Comment by deads2k** [comment]
Clean rebase to confirm no conflicts with new changes.

**Comment by wojtek-t** [comment]
@deads2k - can you please rebase?

**Comment by deads2k** [comment]
already included in https://github.com/kubernetes/kubernetes/pull/17327

**Pull request closed without merging**